starter_code/public/javascript/index.js

```diff
@@ -1,23 +1,89 @@
-const charactersAPI = new APIHandler("http://localhost:8000")
-
-$(document).ready( () => {
-  document.getElementById('fetch-all').onclick = function(){
-
-  }
-  
-  document.getElementById('fetch-one').onclick = function(){
-    
-  }
-  
-  document.getElementById('delete-one').onclick = function(){
-        
-  }
-  
-  document.getElementById('edit-character-form').onsubmit = function(){
-            
-  }
-  
-  document.getElementById('new-character-form').onsubmit = function(){
-                
-  }
-})
+const charactersAPI = new APIHandler("http://localhost:8000");
+
+const charactersInfo = {
+  name: String,
+  occupation: String,
+  debt: Boolean,
+  weapon: String
+};
+
+$(document).ready(() => {
+  document.getElementById("fetch-all").onclick = function() {
+    charactersAPI.getFullList().then(res => {
+      $(".characters-container").empty();
+
+      res.forEach(element => {
+        $(".characters-container").append(
+          `<div class="character-info">
+          <div class="id">Id : ${element.id}</div>
+          <div class="name">Character Name : ${element.name}</div>
+          <div class="occupation">Character Occupation : ${
+            element.occupation
+          }</div>
+          <div class="debt">Character Debt : ${element.debt}</div> 
+          <div class="weapon">Character Weapon :  ${element.weapon}</div>
+          </div>`
+        );
+      });
+    });
+  };
+
+  // struggle to access the value of the object send generate after the promise, so can't display anything on the HTML
+  document.getElementById("fetch-one").onclick = function() {
+    let id = $("#fetch-id").val();
+    console.log(charactersAPI.getOneRegister(id));
+
+    charactersAPI.getOneRegister(id).then(
+      res => {
+        //$(".characters-container").empty();
+        $(".characters-container").append(
+          `<div class="character-info">
+          <div class="name">Character Name : ${this.name}</div>
```

Comment:
you mean `res.name` since the `res` object is holding your response from the server. that's why you're passing it as an argument to the callback. You could also more semantically name it `character` already